File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot/ShooterSubsystems/FlywheelSubsystem.java
package org.firstinspires.ftc.teamcode.Robot.ShooterSubsystems;

import com.acmerobotics.dashboard.config.Config;

import org.firstinspires.ftc.robotcore.internal.hardware.android.GpioPin;

import dev.nextftc.control.ControlSystem;
import dev.nextftc.control.KineticState;
import dev.nextftc.core.subsystems.Subsystem;
import dev.nextftc.ftc.ActiveOpMode;
import dev.nextftc.hardware.controllable.MotorGroup;
import dev.nextftc.hardware.impl.MotorEx;

public class FlywheelSubsystem implements Subsystem {
    public static FlywheelSubsystem INSTANCE = new FlywheelSubsystem();
    public FlywheelSubsystem() { }

    public MotorEx fwl = new MotorEx("fwl");
    public MotorEx fwr = new MotorEx("fwr");
    public MotorGroup fwMotors = new MotorGroup(fwr, fwl);

    public void autoFlywheel(double velocity) {
        fwr.setPower(velocity+0.05);
        ActiveOpMode.telemetry().addData("power2", fwMotors.getPower());
    }

    @Override
    public void periodic() {

    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Master/System/ShooterSubsystems/FlywheelSubsystem.kt
package org.firstinspires.ftc.teamcode.Master.System.ShooterSubsystems

import dev.nextftc.control.ControlSystem
import dev.nextftc.control.KineticState
import dev.nextftc.core.subsystems.Subsystem
import dev.nextftc.hardware.controllable.MotorGroup
import dev.nextftc.hardware.impl.MotorEx
import org.firstinspires.ftc.teamcode.Util.FlywheelState
import kotlin.properties.Delegates

object FlywheelSubsystem: Subsystem {
    val leftFlywheelMotor: MotorEx = MotorEx("fwl")
    val rightFlywheelMotor: MotorEx = MotorEx("fwr")
    val flywheelMotorGroup: MotorGroup = MotorGroup(rightFlywheelMotor,leftFlywheelMotor)

    val flywheelControl: ControlSystem = ControlSystem.builder()
        .velPid(0.01,0.0,0.0)
        .basicFF(0.000325,0.0,0.063)
        .build()

    var currentFlywheelVelocity: Double by Delegates.notNull()
    var flywheelAutoAim: Boolean = false

    fun flywheelAutoAimToggle() {
        flywheelAutoAim = !flywheelAutoAim
    }
    fun setFlywheelVelocity(velocity: Double) {
        if (flywheelAutoAim) {
            flywheelControl.goal = KineticState(0.0,velocity)
            currentFlywheelVelocity = velocity
        }
    }
    fun getFlywheelState(): FlywheelState {
        return FlywheelState(currentFlywheelVelocity, flywheelAutoAim)
    }

    override fun periodic() {
        if (flywheelAutoAim) {
            flywheelMotorGroup.power = flywheelControl.calculate(flywheelMotorGroup.state)
        } else {
            flywheelMotorGroup.power = 0.25
        }
    }
}